examples/js/utils/ldraw/packLDrawModel.js

```diff
@@ -0,0 +1,300 @@
+}
+packedContent += "\n";
+
+// Save output file
+var outPath = fileName + "_Packed.mpd";
+console.log( 'Writing "' + outPath + '"...' );
+fs.writeFileSync( outPath, packedContent );
+
+console.log( 'Done.' );
+
+
+//
+
+function parseObject( fileName, isRoot ) {
+
+	// Returns the located path for fileName or null if not found
+
+	console.log( 'Adding "' + fileName + '".' );
+
+	var originalFileName = fileName;
+
+	var prefix = "";
+	var objectContent = null;
+	for ( var attempt =  0; attempt < 2; attempt ++ ) {
+
+		prefix = "";
+
+		if ( attempt === 1 ) {
+
+			fileName = fileName.toLowerCase();
+
+		}
+
+		if ( fileName.startsWith( '48/' ) ) {
+
+			prefix = "p/";
+
+		}
+		else if ( fileName.startsWith( 's/' ) ) {
+
+			prefix = "parts/";
+
+		}
+
+		var absoluteObjectPath = path.join( ldrawPath, fileName );
+
+		try {
+
+			objectContent = fs.readFileSync( absoluteObjectPath, { encoding: "utf8" } );
+			break;
+
+		}
+		catch ( e ) {
+
+			prefix = "parts/";
+			absoluteObjectPath = path.join( ldrawPath, prefix, fileName );
+
+			try {
+
+				objectContent = fs.readFileSync( absoluteObjectPath, { encoding: "utf8" } );
+				break;
+
+			}
+			catch ( e ) {
+
+				prefix = "p/";
+				absoluteObjectPath = path.join( ldrawPath, prefix, fileName );
+
+				try {
+
+					objectContent = fs.readFileSync( absoluteObjectPath, { encoding: "utf8" } );
+					break;
+
+				}
+				catch ( e ) {
+
+					try {
+
+						prefix = "models/";
+						absoluteObjectPath = path.join( ldrawPath, prefix, fileName );
+
+						objectContent = fs.readFileSync( absoluteObjectPath, { encoding: "utf8" } );
+						break;
+
+					}
+					catch ( e ) {
+						if ( attempt === 1 ) {
+
+							// The file has not been found, add to list of not found
+							listOfNotFound.push( originalFileName );
+
+						}
+
+					}
+
+				}
+
+			}
+
+		}
+
+	}
+
+	var objectPath = path.join( prefix, fileName );
+
+	if ( ! objectContent ) {
+
+		// File was not found, but could be a referenced embedded file.
+		return null;
+
+	}
+
+	if ( objectContent.indexOf( '\r\n' ) !== - 1 ) {
+
+		// This is faster than String.split with regex that splits on both
+		objectContent= objectContent.replace( /\r\n/g, '\n' );
+
+	}
+
+	var processedObjectContent = isRoot ? "" : "0 FILE " + objectPath + "\n";
+
+	var lines = objectContent.split( "\n" );
+
+	for ( var i = 0, n = lines.length; i < n; i ++ ) {
+
+		var line = lines[ i ];
+		var lineLength = line.length;
+
+		// Skip spaces/tabs
+		var charIndex = 0;
+		while ( ( line.charAt( charIndex ) === ' ' || line.charAt( charIndex ) === '\t' ) && charIndex < lineLength ) {
+
+			charIndex++;
+
+		}
+		line = line.substring( charIndex );
+		lineLength = line.length;
+		charIndex = 0;
+
+
+		if ( line.startsWith( '0 FILE ') ) {
+
+			if ( i === 0 ) {
+
+				// Ignore first line FILE meta directive
+				continue;
+
+			}
+
+			// Embedded object was found, add to path map
+
+			var subobjectFileName = line.substring( charIndex ).trim().replace( "\\", "/" );
```

Comment:
lgtm complains about this line, see https://lgtm.com/projects/g/mrdoob/three.js/snapshot/1b5b8ba44b5546eff6c778e54dcc9aa0474ff6b6/files/utils/packLDrawModel.js#xa727a310bfdd391d:1

Any chances to rewrite this code? Is it safe to use `.replace( /\\/g, '/' )`;

Reply:
Yes, I assume it is safe to replace all occurrences of '\\'. Since all filesystem access (i.e. calls to readFileSync) are made relative (prepended with './')

I've made a PR with your proposed change.

Reply:
Fixed with PR #19046